src/network/networkd-route.c

```diff
@@ -1723,12 +1723,16 @@ int manager_rtnl_process_route(sd_netlink *rtnl, sd_netlink_message *message, Ma
                 return 0;
         }
 
-        r = sd_rtnl_message_route_get_table(message, &table);
+        r = sd_netlink_message_read_u32(message, RTA_TABLE, &tmp->table);
+        if (r == -ENODATA) {
+                r = sd_rtnl_message_route_get_table(message, &table);
+                if (r >= 0)
+                        tmp->table = table;
```

Comment:
Hmm, this would be necessary if `sd_rtnl_message_route_get_table()` would set the output param on failure. It shouldn't and doesn't afaict. So why not just:
```C
r = sd_netlink_message_read_u32(message, RTA_TABLE, &tmp->table);
if (r == -ENODATA)
     r = sd_rtnl_message_route_get_table(message, &tmp->table);
if (r < 0)
     ...
```

Reply:
`sd_rtnl_message_route_get_table()` stores result into `uint8_t`, while RTA_TABLE is uint32_t.